### PR TITLE
Fix varSyncEnabled for product variations with removed categories

### PR DESCRIPTION
**Pull request by adrianostanley**
# Summary

This PR will fix an issue with product variations and excluded categories from syncing.

If you had a product with a few variations and **Facebook sync = Do not sync** in all of them, the product update could be blocked if its category was marked as excluded from syncing.

### Setup

1. Create a product with two or more variations
2. Give each variation a price and set **Facebook sync** to **Do not sync**
3. Associate this product with a category

### Steps

1. Go to WooCommerce > Facebook
2. Click on the Product sync tab
3. Exclude the category you used in the setup process
4. Go back to the product admin page (make sure it's refreshed after excluding its category from sync)
5. Click on the Update button
    - [x] The product is updated
6. Pick a variation and set its **Facebook sync** to **Sync and show in catalog**
7. Click on the Update button
    - [ ] The modal below is shown

image

## Before merge

- [x] I have confirmed these changes in each supported minor WooCommerce version

### CODE REVIEW

**Comment by adrianostanley** [comment]
@unfulvio requested your revision since you reviewed the other PR I created which would be released with 2.1.0.

Now this one is set as a 2.0.5 hotfix release.

**Review by unfulvio — APPROVED** [review]
hey @adrianostanley this looks ok! thanks for updating it - I did not merge this yet after reviewing: FB 2.0.4 is out though and this PR seems targeting 2.0.3 branch. I want to make sure it goes in the right branch, which should be #1630, correct?

File: changelog.txt
Comment (wvega) [reply changelog.txt]: @unfulvio since @adrianostanley is out until Wednesday, I applied this suggestion (5984856) and I'm going to merge.

**Pull request merged**
